include/vulkan/vulkan_core.h

```diff
@@ -46,15 +46,36 @@ extern "C" {
 #define VK_HEADER_VERSION 82
 
 
-#define VK_NULL_HANDLE 0
+#ifdef __cplusplus
+    struct Null_t{ explicit operator uint64_t() const { return 0; } };
+    #define VK_NULL_HANDLE Null_t{}
```

Comment:
Could this be `std::nullptr_t`?

Reply:
Well, it is semantically a different thing. Handle is not a pointer. Besides pointers are platform dependent things. Non-dispatchable Vulkan handles are 64-bit integral values.